server/src/apis/openalex/openalex.ts

```diff
@@ -0,0 +1,57 @@
+import axios from 'axios';
+import { IPaper } from '../../models/Paper';
+import { Author } from '../../services/search.types';
+import { paperFromWork, authorFromAuthorResponse } from './converters';
+import { AuthorsResponse, Work, WorksResponse } from './types';
+
+const OPEN_ALEX_URL = 'https://api.openalex.org';
+const MAILTO = 'paperaweek@protonmail.com';
```

Comment:
This got me thinking, do you think it's worth creating a generic `openAlexGet` that handles MAILTO and other "always on" params? Something like
```ts
/**
 *
 * @param entity the entity you want from openAlex (e.g. works, authors)
 * @param parameters a dictionary of the parameters you want to search for
 * @param limit limits your query to MAX_RESULTS if true
 * @returns a Promise fulfilling the generic or null
 */
async function openAlexGet<T>(
  entity: string,
  parameters: Record<string, string> = {},
  limit = false
): Promise<T | null> {
  try {
    const response = await axios.get<T>(`${OPEN_ALEX_URL}/${entity}`, {
      params: {
        ...parameters,
        ...(limit ? { per_page: MAX_RESULTS } : undefined),
        mailto: MAILTO,
      },
    });

    if (response.data) {
      return response.data;
    }

    console.error(`Fetching ${entity} returned no data`);
    return null;
  } catch (err) {
    if (err instanceof Error) {
      console.error(`Failed to get ${entity}, error: ${err.message}`);
      console.error(err);
    }

    return null;
  }
}
```

Usage:
```ts
export async function getAuthorByName(name: string): Promise<Author | null> {
  const authorResponse = await openAlexGet<AuthorsResponse>('authors', {
    filter: `display_name.search:${name}`,
    sort: 'cited_by_count',
  });
  if (!authorResponse) {
    return null;
  }
  return authorFromAuthorResponse(authorResponse);
}
```